cmd/format-disk-cache.go

```diff
@@ -146,32 +147,25 @@ func initFormatCache(drives []string) (formats []*formatCacheV1, err error) {
 	return nformats, nil
 }
 
-func loadFormatCache(drives []string) (formats []*formatCacheV1, err error) {
-	var errs []error
-	for _, drive := range drives {
-		cacheFormatPath := pathJoin(drive, formatConfigFile)
-		f, perr := os.Open(cacheFormatPath)
-		if perr != nil {
-			formats = append(formats, nil)
-			errs = append(errs, perr)
-			continue
+func loadFormatCache(drives []string) ([]*formatCacheV1, error) {
+	formats := make([]*formatCacheV1, len(drives))
+	for i, drive := range drives {
+		cacheFormatPath := pathJoin(drive, minioMetaBucket, formatConfigFile)
+		f, err := os.Open(cacheFormatPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return nil, err
 		}
 		defer f.Close()
-		format, perr := formatMetaCacheV1(f)
-		if perr != nil {
-			// format could not be unmarshalled.
-			formats = append(formats, nil)
-			errs = append(errs, perr)
+		format, err := formatMetaCacheV1(f)
+		if err != nil {
 			continue
 		}
-		formats = append(formats, format)
-	}
-	for _, perr := range errs {
-		if perr != nil {
-			err = perr
-		}
```

Comment:
this code should not be refactored out. when one or more drives in the cache configuration could not be initialized, the error message is not being reported on console.  
``` 
export MINIO_CACHE_DRIVES="/home/k/cache/drive1;/drive2"
```
/drive2 is invalid, but server starts without any indication to user that one of the drives is unusable.

Reply:
This is the case where one of the disks had failed and was unmounted. i.e loadFormatCache() should not return error.

But yes we should log the error even if we dont return the error so that the user knows that there is something wrong with the particular drive entry. Kannappan is adding the logging messages in the error logging PR